frontend/simple/js/backend/hapi.js

```diff
@@ -105,12 +109,17 @@ export class HapiNamespace extends TrustedNamespace {
   async register (name: string, value: string) {
     console.log(`registering name:`, name)
     let response = await request.post(`${process.env.API_URL}/name`).send({name, value})
+    store.dispatch('addUser', {username: name, identity: value})
     return response.body.data
   }
   async lookup (name: string) {
     console.log(`lookup name:`, name)
     // TODO: should `name` be encodeURI'd?
+    if (store.state.users[name]) { return store.state.users[name] }
     let response = await request.get(`${process.env.API_URL}/name/${name}`)
+    store.dispatch('addUser', {username: name, identity: response.body.data.value})
```

Comment:
You'll have to clarify what you want then. What does it matter if it modifies the state? Most of the functions in this file have this same property and are equally unclear.